lib/grainstore/mml-builder/mml-builder.js

```diff
@@ -55,28 +57,66 @@ MMLBuilder.prototype.toXML = function(callback) {
     }
 
     debug('Acquiring child process');
-    childMMLBuilderPool.acquire(function(err, child) {
+    childMMLBuilderPool.acquire(function (err, child) {
         if (err) {
             return callback(new Error('Unable to generate Mapnik XML'));
-        } else {
-            debug('Waiting for Mapnik XML');
-            child.once('message', function (result) {
+        }
+
+        debug('Waiting for Mapnik XML');
+
+        function done (result) {
+            debug('Child process sent a result back to the parent');
+            if (timeout) {
+                clearTimeout(timeout);
+            }
+
+            if (result.err && result.err.includes('ENOMEM')) {
+                childMMLBuilderPool.destroy(child);
+                process.emit('ENOMEM');
+
+                return callback(new Error(result.err));
+            }
+
```

Comment:
This change introduces 3 different calls to `childMMLBuilderPool.release(child);` vs 1. Can't you just call release after the ENOMEM check?

Reply:
done